### PR TITLE
Pop up for GUI persistance of uncommitted state

### PR DESCRIPTION
**Pull request by anthonydillon**
Displays a browser pop up when closing or refreshing the window with uncommitted changes.

### CODE REVIEW

**Comment by anthonydillon** [comment]
## QA
- Load the GUI in sandbox and GIJoe
- Check that you and refresh and close the windows with no changes
- Add a service to the canvas and then try and refresh and close
- You should get a pop up warning you that you will lose your changes

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5796/
Test PASSed.

File: jujugui/static/gui/src/test/test_utils.js [hunk 1]
```diff
@@ -721,6 +721,41 @@ describe('utilities', function() {
...
+
+    it('does not block when no uncommitted changes', function() {
+      var context = {env: {
+        get: sinon.stub().returns({
+          getCurrentChangeSet: sinon.stub().returns({})
+        })
+      }};
+
+      var result = utils.unloadWindow.call(context);
+      assert.equal(result, undefined);
```
Comment (hatched) [inline jujugui/static/gui/src/test/test_utils.js]: this should probably use `strictEqual`

File: jujugui/static/gui/src/app/views/utils.js [hunk 2]
```diff
@@ -1223,6 +1223,21 @@ YUI.add('juju-view-utils', function(Y) {
...
   };
 
+  /**
+    Displays a confirmation when closing window if there are uncommitted
+    changes
+
+    @method unloadWindow
+    @param {Object} env Reference to the app env.
+  */
+  utils.unloadWindow = function() {
```
Comment (hatched) [inline jujugui/static/gui/src/app/views/utils.js]: What do you think about simply passing the ecs instance into this method?

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5798/
Test PASSed.

File: jujugui/static/gui/src/test/test_utils.js (hunk 1 above)
Comment (anthonydillon) [inline jujugui/static/gui/src/test/test_utils.js]: Done, thanks/

**Comment by kadams54** [comment]
Code looks good. :+1: thanks!

**Comment by kadams54** [comment]
QA is :ok:.

**Comment by kadams54** [comment]
QA is :ok:.

**Comment by hatched** [comment]
Thanks @anthonydillon 
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**